### PR TITLE
Feature/prevalence risk effects

### PR DESCRIPTION
**Pull request by halfhorst**
adds a risk effect intended for prevalence that is correlated with household tb exposure. this relies on probability of exposure to household tb, the relative risk of hhtb, and the PAF of hhtb.

I opted to pre-compute the data nad put it in the artifact. The RR data is the same for all measures and entities, so I duplicated it again for these new entities and measures. the paf is computed directly from rr for all entities and measures present

### CODE REVIEW

File: src/vivarium_csu_ltbi/components/risk.py [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+        self.exposed_exposure_probability = self._get_exposed_exposure_probability_data(builder)
+        self.exposure = builder.value.get_value(f'{self.risk.name}.exposure')
+        self.population_attributable_fraction = self._get_population_attributable_fraction_data(builder)
+
+        builder.value.register_value_modifier(f'{self.target.name}.{self.target.measure}',
+                                              modifier=self.adjust_target,
+                                              requires_values=[f'{self.risk.name}.exposure'],
+                                              requires_columns=['age', 'sex'])
+
+    def _get_relative_risk_date(self, builder):
```
Comment (KjellSwedin) [inline src/vivarium_csu_ltbi/components/risk.py]: Is this intended or a typo: 
_get_relative_risk_date
or
_get_relative_risk_data

**Review by KjellSwedin — APPROVED** [review]

File: src/vivarium_csu_ltbi/components/disease.py [hunk 2]
```diff
@@ -11,7 +18,193 @@ def wrap_data_getter(id):
     }
 
 
-class BetterDiseaseModel(DiseaseModel):
+class BetterDiseaseModel(Machine):
+    """FIXME: This class used to extend DiseaseModel itself, but the population
```
Comment (halfhorst) [inline src/vivarium_csu_ltbi/components/disease.py:22]: direct copy of DiseaseModel + additional value dependencies on the population initializer (hardcoded ..)

File: src/vivarium_csu_ltbi/components/risk.py (hunk 1 above)
Comment (halfhorst) [reply src/vivarium_csu_ltbi/components/risk.py]: not intended :sweat_smile:

File: src/vivarium_csu_ltbi/components/risk.py [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+        return exposure
+
+    def _get_population_attributable_fraction_data(self, builder):
+        paf_data = get_population_attributable_fraction_data(builder, self.risk, self.target, self.randomness)
+        paf = builder.lookup.build_table(paf_data, key_columns=['sex'],
+                                         parameter_columns=['age', 'year'])
+        return paf
+
+    def adjust_target(self, index, target):
+        is_exposed = self.exposure(index) == 'cat1'
```
Comment (collijk) [inline src/vivarium_csu_ltbi/components/risk.py]: Fixes necessary as per slack convo.  

If you have ones in the appropriate spots in your rr data, you should just be able to do 
```
exposure = self.exposure(index)
rr = self.relative_risk(index)
paf = self.population_attributable_fraction(index)
target *= (1 - paf) * rr.lookup(exposure.index, exposure)
```

File: src/vivarium_csu_ltbi/components/disease.py (hunk 2 above)
Comment (collijk) [reply src/vivarium_csu_ltbi/components/disease.py:22]: Let's take 15 mins monday and talk about how to incorporate this fix into vph.

**Review by collijk — COMMENTED** [review]
Make the adjust target fix, and replace string constants with references to globals, then good to merge.

File: src/vivarium_csu_ltbi/components/risk.py (hunk 3 above)
Comment (halfhorst) [reply src/vivarium_csu_ltbi/components/risk.py]: good point

**Pull request merged**